metrics/sources/kubelet/summary.go

```diff
@@ -0,0 +1,342 @@
+		LabeledMetrics: []LabeledMetric{},
+	}
+	containerMetrics.Labels[LabelMetricSetType.Key] = MetricSetTypePodContainer
+	containerMetrics.Labels[LabelContainerName.Key] = container.Name
+
+	startTime := container.StartTime.Time
+	this.decodeUptime(containerMetrics, startTime)
+	this.decodeCPUStats(containerMetrics, container.CPU, startTime)
+	this.decodeMemoryStats(containerMetrics, container.Memory, startTime)
+	this.decodeFsStats(containerMetrics, RootFsKey, container.Rootfs)
+	this.decodeFsStats(containerMetrics, LogsKey, container.Logs)
+	this.decodeUserDefinedMetrics(containerMetrics, container.UserDefinedMetrics)
+
+	return containerMetrics
+}
+
+func (this *summaryMetricsSource) decodeUptime(metrics *MetricSet, startTime time.Time) {
+	if startTime.IsZero() {
+		return
+	}
+
+	uptime := uint64(time.Since(startTime).Nanoseconds() / time.Millisecond.Nanoseconds())
+	this.addIntMetric(metrics, &MetricUptime, &uptime, startTime)
+}
+
+func (this *summaryMetricsSource) decodeCPUStats(metrics *MetricSet, cpu *stats.CPUStats, startTime time.Time) {
+	if cpu == nil {
+		return
+	}
+
+	this.addIntMetric(metrics, &MetricCpuUsage, cpu.UsageCoreNanoSeconds, startTime)
+
+	// Need to translate from nanocores to millicores.
+	if cpu.UsageNanoCores != nil {
+		const nanoPerMilli uint64 = 1e6
+		cores := *cpu.UsageNanoCores / nanoPerMilli
+		this.addIntMetric(metrics, &MetricCpuUsageRate, &cores, startTime)
+	}
+}
+
+func (this *summaryMetricsSource) decodeMemoryStats(metrics *MetricSet, memory *stats.MemoryStats, startTime time.Time) {
+	if memory == nil {
+		return
+	}
+
+	this.addIntMetric(metrics, &MetricMemoruUsage, memory.UsageBytes, startTime)
+	this.addIntMetric(metrics, &MetricMemoryWorkingSet, memory.WorkingSetBytes, startTime)
+	this.addIntMetric(metrics, &MetricMemoryPageFaults, memory.PageFaults, startTime)
+	this.addIntMetric(metrics, &MetricMemoryMajorPageFaults, memory.MajorPageFaults, startTime)
+}
+
+func (this *summaryMetricsSource) decodeNetworkStats(metrics *MetricSet, network *stats.NetworkStats, startTime time.Time) {
+	if network == nil {
+		return
+	}
+
+	this.addIntMetric(metrics, &MetricNetworkRx, network.RxBytes, startTime)
+	this.addIntMetric(metrics, &MetricNetworkRxErrors, network.RxErrors, startTime)
+	this.addIntMetric(metrics, &MetricNetworkTx, network.TxBytes, startTime)
+	this.addIntMetric(metrics, &MetricNetworkTxErrors, network.TxErrors, startTime)
+}
+
+func (this *summaryMetricsSource) decodeFsStats(metrics *MetricSet, fsKey string, fs *stats.FsStats) {
+	if fs == nil {
+		return
+	}
+
+	fsLabels := map[string]string{LabelResourceID.Key: fsKey}
+	if fs.UsedBytes != nil {
+		usage := LabeledMetric{
+			Name:       MetricFilesystemUsage.Name,
+			Labels:     fsLabels,
+			ValueType:  ValueInt64,
+			MetricType: MetricFilesystemUsage.Type,
+			IntValue:   int64(*fs.UsedBytes),
+		}
+		metrics.LabeledMetrics = append(metrics.LabeledMetrics, usage)
+	}
+
+	if fs.CapacityBytes != nil {
```

Comment:
Done. Re: schema - It looks like that schema is out of date (it's missing a handful of metrics). I can update it in a separate PR, but I think it would ideally be generated from the metrics definitions.